### PR TITLE
Add color info

### PR DESCRIPTION
**Pull request by davidtcalabrese**

### CODE REVIEW

File: haggis.js [hunk 1]
```diff
@@ -1,161 +1,187 @@
 async function parseLogAndPopulateForm() {
-    const stats = parseLog(document.getElementById('logText').value);
-    const data = await serializeJson(stats);
-    document.getElementById('data').value = data;
-    document.getElementById('statsForm').submit();
+  const stats = parseLog(document.getElementById("allLogs").textContent);
+  const data = await serializeJson(stats);
+  document.getElementById("data").value = data;
```
Comment (pwinckles) [inline haggis.js:4]: Out of curiosity, why did you change all of the `'` to `"`? Is the single quote not appropriate?

File: haggis.js [hunk 2]
```diff
@@ -1,161 +1,187 @@
...
+      if (!(player in game.playerStats)) {
+        game.playerStats[player] = createPlayerStats();
+      }
+      game.playerStats[player].led += 1;
+      currentRound = {
+        startPlayer: player,
+        bets: {},
+        tens: {},
+        bombs: {},
+        sequenceBombs: {},
```
Comment (pwinckles) [inline haggis.js]: can you rename this to `colorBombs`?

File: haggis.js [hunk 3]
```diff
@@ -1,161 +1,187 @@
...
-            currentRound.bets[player] = words[2];
-            game.playerStats[player].bets[words[2]] = (game.playerStats[player].bets[words[2]] ?? 0) + 1;
-            game.playerStats[player].totalBets += 1;
-            continue;
+        currentRound.sums[player] = (currentRound.sums[player] ?? 0) + num;
+        game.playerStats[player].sumTotal += num;
+      }
+
+      if (isBomb(words[2])) {
+        if (isSequenceBomb(words[2])) {
```
Comment (pwinckles) [inline haggis.js]: same here `isColorBomb`

File: haggis.js [hunk 4]
```diff
@@ -1,161 +1,187 @@
...
-            game.rounds.push(currentRound);
-            continue;
-        }
+    const score = line.match(/scores (\d+) point/);
+    if (score) {
+      game.playerStats[player].score += Number(score[1]);
+      currentRound.points[player] =
+        (currentRound.points[player] ?? 0) + Number(score[1]);
+    }
+    if (words[1].includes("bets")) {
```
Comment (pwinckles) [inline haggis.js:77]: Can you add a new line between 76 and 77?

File: haggis.js [hunk 5]
```diff
@@ -176,228 +202,343 @@ function parseLog(text) {
...
+  output += `    <td>${player1Stats.ledAndWon}</td>\n`;
+  output += `    <td>${player2Stats.ledAndWon}</td>\n`;
+  output += "  </tr>\n";
+  output += "  <tr>\n";
+  output += "    <td>10s</td>\n";
+  output += `    <td>${player1Stats.tens}</td>\n`;
+  output += `    <td>${player2Stats.tens}</td>\n`;
+  output += "  </tr>\n";
+  output += "  <tr>\n";
+  output += "    <td>Sequence Bombs</td>\n";
```
Comment (pwinckles) [inline haggis.js]: "Color Bombs"

File: haggis.js [hunk 6]
```diff
@@ -1,161 +1,187 @@
...
+      inLog = true;
+      if (!(player in game.playerStats)) {
+        game.playerStats[player] = createPlayerStats();
+      }
+      game.playerStats[player].led += 1;
+      currentRound = {
+        startPlayer: player,
+        bets: {},
+        tens: {},
+        bombs: {},
```
Comment (pwinckles) [inline haggis.js]: I think we can remove the `bombs` attribute

File: haggis.js [hunk 7]
```diff
@@ -176,228 +202,343 @@ function parseLog(text) {
...
-       ledAndWon: 0,
-       sumTotal: 0,
-       sumMin: 0,
-       sumMax: 0,
-       sumAvg: 0,
-       largerSum: 0,
-   };
+  return {
+    tens: 0,
+    bombs: 0,
```
Comment (pwinckles) [inline haggis.js:215]: Same here, we don't need `bombs` any more

File: haggis.js [hunk 8]
```diff
@@ -176,228 +202,343 @@ function parseLog(text) {
...
+    output += `    <td>${round.bets[player2] ?? "NA"}</td>\n`;
     output += "  <tr>\n";
     output += "    <td>10s</td>\n";
-    output += `    <td>${player1Stats.tens}</td>\n`;
-    output += `    <td>${player2Stats.tens}</td>\n`;
+    output += `    <td>${round.tens[player1] ?? 0}</td>\n`;
+    output += `    <td>${round.tens[player2] ?? 0}</td>\n`;
     output += "  </tr>\n";
     output += "  <tr>\n";
     output += "    <td>Rainbow/Color Bombs</td>\n";
```
Comment (pwinckles) [inline haggis.js]: We don't need the `Rainbow/Color Bombs` line any more

File: haggis.js [hunk 9]
```diff
@@ -176,228 +202,343 @@ function parseLog(text) {
...
+
+function stripColorFromCard(inputString) {
+  if (/^[rbpy]/.test(inputString)) {
+    return inputString.slice(1);
+  }
+  return inputString;
+}
+
+function isSequenceBomb(inputString) {
+  const regex = /^(?:([rpyb])3)-\1?5-\1?7-\1?9$/;
```
Comment (pwinckles) [inline haggis.js:487]: nice

File: haggis.js [hunk 10]
```diff
@@ -1,161 +1,187 @@
...
 }
 
-function parseLog(text) {
-    const lines = text.split('\n');
-    const game = {
-        players: [],
-        rounds: [],
-        playerStats: {},
-    };
+function parseLog(logLines) {
```
Comment (pwinckles) [inline haggis.js:22]: My browser is complaining about a balanced `{` issue and I can run the code.

```
Uncaught SyntaxError: missing } after function body[haggis.js:545:1](file:///home/pwinckles/workspace/haggis-stats/haggis.js)note: { opened at line 22, column 29
```

File: haggis.js (hunk 1 above)
Comment (davidtcalabrese) [reply haggis.js:4]: I think this was prettier.

File: haggis.js (hunk 10 above)
Comment (davidtcalabrese) [reply haggis.js:22]: fixed
